specification/deploymentmanager/resource-manager/Microsoft.DeploymentManager/preview/2019-11-01-preview/deploymentmanager.json

```diff
@@ -0,0 +1,2465 @@
+          "example": "userAssigned"
+        },
+        "identityIds": {
+          "type": "array",
+          "items": {
+            "type": "string",
+            "example": "/subscriptions/mySubId/resourcegroups/myRG/providers/Microsoft.ManagedIdentity/userassignedidentities/myIdentity"
+          },
+          "description": "The list of identities."
+        }
+      },
+      "required": [
+        "type",
+        "identityIds"
+      ]
+    },
+    "RolloutRequest": {
+      "description": "Defines the PUT rollout request body.",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "properties": {
+        "identity": {
+          "description": "Identity for the resource.",
+          "$ref": "#/definitions/Identity"
+        },
+        "properties": {
+          "description": "Defines the properties that make up a rollout request.",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/RolloutRequestProperties"
+        }
+      },
+      "required": [
+        "identity",
+        "properties"
+      ]
+    },
+    "RolloutRequestProperties": {
+      "description": "The properties for defining a rollout.",
+      "properties": {
+        "buildVersion": {
+          "type": "string",
+          "description": "The version of the build being deployed.",
+          "example": "1.0.0.1"
+        },
+        "artifactSourceId": {
+          "type": "string",
+          "description": "The reference to the artifact source resource Id where the payload is located.",
+          "example": "Microsoft.DeploymentManager/artifactSources/payloadStore1"
+        },
+        "targetServiceTopologyId": {
+          "type": "string",
+          "description": "The resource Id of the service topology from which service units are being referenced in step groups to be deployed.",
+          "example": "Microsoft.DeploymentManager/serviceTopologies/myServiceTopology"
+        },
+        "stepGroups": {
+          "description": "The list of step groups that define the orchestration.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Step"
+          }
+        }
+      },
+      "required": [
+        "stepGroups",
+        "targetServiceTopologyId",
+        "buildVersion"
+      ]
+    },
+    "Step": {
+      "description": "The properties that define an Azure Deployment Manager step.",
+      "properties": {
+        "name": {
+          "description": "The name of the step group.",
+          "type": "string"
+        },
+        "dependsOnStepGroups": {
+          "description": "The list of step group names on which this step group depends on.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "example": [
+            "stepGroup1",
+            "stepGroup2"
+          ]
+        },
+        "preDeploymentSteps": {
```

Comment:
is DeploymentStep same concept of Step here? little confusing on un-consistent.

Reply:
Renamed this type to StepGroup, which is what we originally intended. The StepGroup contains references to multiple steps that are created separately.